Sources/PackageModel/Toolset.swift

```diff
@@ -76,6 +76,15 @@ extension Toolset {
             )
         }
 
+        let rootPaths: [AbsolutePath]
+        if let rootPath = decoded.rootPath {
```

Comment:
nit: can use `map`?

Reply:
We can, not sure if this is more readable?

```swift
        let rootPaths = try decoded.rootPath.map {
            [
                try AbsolutePath(validating: $0, relativeTo: toolsetPath.parentDirectory)
            ]
        } ?? []
```

I was initially a proponent of `map` everywhere for optionals due to my FP background, but eventually came to prefer `if let` from a readability perspective, and it's not much more verbose if at all.

Reply:
I'd love to use SE-0380 (`if` expression) here eventually when that becomes available though.